### DISCUSSION TITLE
Can we both assert that an array is not empty and assert his type ?

### DISCUSSION THREAD

**Discussion by TomChv** [cue-lang/cue#1175] [Q&A]
Hi!

Is it possible to use both `list.MinItems()` and a type assertion?

I would like to do something like this but it doesn't work...

```cue
import (
    "list"
    "strings"
)

#ListMin: {
    arr: list.MinItems(1)
}

// Assert list isn't empty
arrMinNumber: #ListMin & { arr: [1] }
arrMinString: #ListMin & { arr: ["a"] } // I want that to throw error
//arrMinErr: #ListMin & { arr: [] } // Fail because list shouldn't be empty


#ListNumber: {
    arr: [...number]
}

// Assert list is type of number
arrNum: #ListNumber & { arr: [1, 2] }
arrNumEmpty: #ListNumber & { arr: [] } // I want that to throw error
//arrNumString: #ListNumber & { arr: ["a"] } // Fail because list should only be composed of string

#ListNumberNotEmpty: {
    arr: [...number] & list.MinItems(1) // doesn't work and instantly throw error
}
```

How can I construct a definition that asserts that I will have a not empty array of type `X`?

**Comment by seh** [comment]
Are you looking for the following?
```cue
arrMinNumber: [number, ...number]
arrMinString: [string, ...string]
```

**Reply by myitcv** [reply]
Welcome to the CUE project, @TomChv!

I transferred your issue to be a discussion; it's not clear to me there is a bug/feature request here.

In addition to @seh's answer, see also the https://pkg.go.dev/cuelang.org/go@v0.4.0/pkg/list package for list-related constraints.
